air/src/trace/rows.rs

```diff
@@ -0,0 +1,349 @@
+
+impl From<RowIndex> for u32 {
+    fn from(step: RowIndex) -> u32 {
+        step.0
+    }
+}
+
+impl From<RowIndex> for u64 {
+    fn from(step: RowIndex) -> u64 {
+        step.0 as u64
+    }
+}
+
+impl From<RowIndex> for usize {
+    fn from(step: RowIndex) -> usize {
+        step.0 as usize
+    }
+}
+
+impl From<RowIndex> for Felt {
+    fn from(step: RowIndex) -> Felt {
+        Felt::from(step.0)
+    }
+}
+
+// INTO ROW INDEX
+// ================================================================================================
+
+/// Converts a usize value into a [`RowIndex`].
+///
+/// # Panics
+///
+/// This function will panic if the number represented by the usize is greater than the maximum
+/// [`RowIndex`] value, `u32::MAX`.
+impl From<usize> for RowIndex {
+    fn from(value: usize) -> Self {
+        let value = u32::try_from(value).map_err(|_| RowIndexError::InvalidSize(value)).unwrap();
+        value.into()
+    }
+}
+
+/// Converts a u64 value into a [`RowIndex`].
+///
+/// # Panics
+///
+/// This function will panic if the number represented by the u64 is greater than the maximum
+/// [`RowIndex`] value, `u32::MAX`.
+impl TryFrom<u64> for RowIndex {
+    type Error = RowIndexError<u64>;
+
+    fn try_from(value: u64) -> Result<Self, Self::Error> {
+        let value = u32::try_from(value).map_err(|_| RowIndexError::InvalidSize(value))?;
+        Ok(RowIndex::from(value))
+    }
+}
+
+impl From<u32> for RowIndex {
+    fn from(value: u32) -> Self {
+        Self(value)
+    }
+}
+
+/// Converts an i32 value into a [`RowIndex`].
+///
+/// # Panics
+///
+/// This function will panic if the number represented by the i32 is less than 0.
+impl From<i32> for RowIndex {
+    fn from(value: i32) -> Self {
+        let value = u32::try_from(value).map_err(|_| RowIndexError::InvalidSize(value)).unwrap();
+        RowIndex(value)
+    }
+}
+
+// ROW INDEX OPS
+// ================================================================================================
+
+/// Subtracts a usize from a [`RowIndex`].
+///
+/// # Panics
+///
+/// This function will panic if the number represented by the usize is greater than the maximum
+/// [`RowIndex`] value, `u32::MAX`.
+impl Sub<usize> for RowIndex {
+    type Output = RowIndex;
+
+    fn sub(self, rhs: usize) -> Self::Output {
+        let rhs = u32::try_from(rhs).map_err(|_| RowIndexError::InvalidSize(rhs)).unwrap();
+        RowIndex(self.0 - rhs)
+    }
+}
+
+impl Sub<RowIndex> for usize {
+    type Output = RowIndex;
+
+    fn sub(self, rhs: RowIndex) -> Self::Output {
+        (self - rhs.0 as usize).into()
+    }
+}
+
+impl Sub<RowIndex> for RowIndex {
+    type Output = RowIndex;
+
+    fn sub(self, rhs: RowIndex) -> Self::Output {
+        RowIndex(self.0 - rhs.0)
+    }
+}
+
+impl Sub<RowIndex> for u32 {
+    type Output = RowIndex;
+
+    fn sub(self, rhs: RowIndex) -> Self::Output {
+        RowIndex(self - rhs.0)
+    }
+}
```

Comment:
Similar comment to the ones above. Generally, the way I was thinking about this was that we could add, subtract, etc. values to/from row indexes, but adding two row indexes or subtracting row index from a row index would not be a "native" operation. But maybe there are enough cases to justify this?

Reply:
This one is unused actually, will remove.

Reply:
@bobbinth could you please clarify - do you want all arithmetic overloads removed from RowIndex type?

Currently there is +, -, *, %, shift right, +=, -=.

Thanks.

Reply:
I think some are fine. For example:

- `RoIndex + u32`, `RoIndex += u32`
- `RoIndex - u32`, `RoIndex -= u32`
- `RoIndex * u32`, `RoIndex *= u32`
- `RoIndex + usize`, `RoIndex += usize`
- `RoIndex - usize`, `RoIndex -= usize`
- `RoIndex * usize`, `RoIndex *= usize`

But things like `RowIndex * RowIndex` or `RowIndex + RowIndex` or bit shifts or modulus operations feel a bit weird.